src/database-layer/db.ts

```diff
@@ -2,12 +2,14 @@
 import * as _mysql from 'mysql'
 import * as _pg from 'pg'
 import * as _pgConnectionString from 'pg-connection-string'
-
+import * as EventEmitter from 'eventemitter3';
 import * as _ from 'lodash'
 import * as Promise from 'bluebird'
 import TypedError = require('typed-error')
 import * as env from '../config-loader/env'
 
+export const metricsEmitter = new EventEmitter();
```

Comment:
I think renaming this to just `metrics` make things flow a little nicer, the `Emitter part feels redundant. eg
```ts
db.metrics.on('db_error', ...)
db.metrics.on('db_query_time', ...)
metrics.emit(...)

Reply:
Nice!